### PR TITLE
feat: add dummy implementation of IndexClientApi to core and http-client

### PR DESCRIPTION
**Pull request by ukstv**
- Move shared types to common
- Add dummy impl to core and http-client

### CODE REVIEW

File: packages/common/src/index-client-api.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import type { StreamID } from '@ceramicnetwork/streamid'
+import type { StreamState } from './stream.js'
+
+/**
+ * Allowed ordering for an indexing query.
+ */
+export enum Ordering {
```
Comment (stbrody) [inline packages/common/src/index-client-api.ts:7]: I don't think this should be exposed in the client API.  As we discussed earlier, we want to always use chronological ordering for now, and avoid exposing any ordering options until we've settled on user-facing names for them.

File: packages/core/src/ceramic.ts [hunk 2]
```diff
@@ -175,11 +177,12 @@ export class Ceramic implements CeramicApi {
   public readonly dispatcher: Dispatcher
   public readonly loggerProvider: LoggerProvider
   public readonly pin: PinApi
+  public readonly index: IndexClientApi
```
Comment (stbrody) [inline packages/core/src/ceramic.ts:180]: `IndexClientApi` sounds to me like what `RemoteIndexApi` actually is.  Can we rename `IndexClientApi` to `IndexApi` and `IndexApi` to `ServerIndexApi` or `LocalIndexApi`?

File: packages/core/src/ceramic.ts [hunk 3]
```diff
@@ -175,11 +177,12 @@ export class Ceramic implements CeramicApi {
...
   readonly repository: Repository
 
   readonly _streamHandlers: HandlersMap
   private readonly _anchorValidator: AnchorValidator
-  private readonly _indexing: DatabaseIndexAPI | undefined
+  private readonly _indexing: DatabaseIndexApi | undefined
```
Comment (stbrody) [inline packages/core/src/ceramic.ts:185]: Can we hide the `DatabaseIndexApi` within the `IndexApi`?  Does the `Ceramic` class need both or can one abstract away the other?

**Review by stbrody — CHANGES_REQUESTED** [review]

File: packages/common/src/index-client-api.ts (hunk 1 above)
Comment (ukstv) [reply packages/common/src/index-client-api.ts:7]: Again, let me remind you about tradeoffs here. It is easier to remove the option than to add it at a later time.

File: packages/core/src/ceramic.ts (hunk 2 above)
Comment (ukstv) [reply packages/core/src/ceramic.ts:180]: good idea.

File: packages/core/src/ceramic.ts (hunk 3 above)
Comment (stbrody) [reply packages/core/src/ceramic.ts:185]: sorry, can you explain the concern a bit more?  Couldn't `LocalIndexApi` have an `init()` method which in turn calls `DatabaseIndexApi::init()`, and Ceramic's `_init()` can just call the `LocalIndexApi::init()`?

Comment (ukstv) [reply packages/core/src/ceramic.ts:185]: That was the first version of the initialisation logic haha :) Maybe it is not a big deal to make `init` available after all.

Comment (ukstv) [reply packages/core/src/ceramic.ts]: Okay, I feel it is better now.

**Review by stbrody — APPROVED** [review]
:shipit:

**Pull request merged**